src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -139,95 +175,131 @@ public static void HashBytes(HashAlgorithm hashAlgorithm, byte[] bytes)
         /// </summary>
         /// <param name="reader">binary reader to zip archive</param>
         /// <returns>metadata with offsets and positions for entries</returns>
-        internal static SignedPackageArchiveMetadata ReadSignedArchiveMetadata(BinaryReader reader)
+        public static SignedPackageArchiveMetadata ReadSignedArchiveMetadata(BinaryReader reader)
         {
-            var metadata = new SignedPackageArchiveMetadata();
+            if (reader == null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+
+            var metadata = new SignedPackageArchiveMetadata()
+            {
+                StartOfFileHeaders = reader.BaseStream.Length,
```

Comment:
I initializes as the length so that it has the maximum possible value. This will be overwritten once the code starts finding file headers and will keep the lowest possible offset it founds (typically 0, but not necessarily)